src/lib/components/HttpClient.ts

```diff
@@ -88,4 +89,20 @@ export class HttpClient {
       return this.DEFAULT_ERROR;
     }
   }
+
+  public async fetch(config: HttpRequest): Promise<HttpResponse> {
```

Comment:
perfect, would be good to get some basic docs down as well.

also DEFAULT_ERROR shows the shape of errors we normally return from API errors. unsure if we need to follow that shape here but looks good!